twisted/web/test/test_web.py

```diff
@@ -136,6 +136,32 @@ def test_makeSession(self):
         self.assertIsInstance(session.uid, bytes)
 
 
+    def test_sessionUIDGeneration(self):
+        """
+        L{site.getSession} generates L{Session} objects with distinct UIDs from
+        a secure source of entropy.
+        """
+        site = server.Site(resource.Resource())
+        # Ensure that we _would_ use the unpredictable random source if the
+        # test didn't stub it.
+        self.assertIdentical(site._entropy, os.urandom)
```

Comment:
It might be nice if this had an assert message about the quality of the entropy source, in case it isn't `os.urandom`. (This suggestion should not block this from being merged.)